OpenLaTeXStudio/EditorModule/src/latexstudio/editor/LaTeXSettingsPanel.java

```diff
@@ -113,20 +158,39 @@ private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRS
         }
     }//GEN-LAST:event_jButton1ActionPerformed
 
+    private void jFormattedTextField1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jFormattedTextField1ActionPerformed
+        // TODO add your handling code here:
+    }//GEN-LAST:event_jFormattedTextField1ActionPerformed
+
     void load() {
         jTextField1.setText(NbPreferences.forModule(LaTeXSettingsPanel.class).get("latexPath", ""));
+        jFormattedTextField1.setText(String.valueOf(NbPreferences.forModule(LaTeXSettingsPanel.class).getInt(LaTeXSettingsPanel.KEY_AUTOCOMPLETE_DELAY, DEFAULT_AUTOCOMPLETE_DELAY_MS)));
     }
-
+    
     void store() {
         NbPreferences.forModule(LaTeXSettingsPanel.class).put("latexPath", jTextField1.getText());
+        
+        int autoCompleteDelay = DEFAULT_AUTOCOMPLETE_DELAY_MS;
+        
+        try{
+            autoCompleteDelay = Integer.parseInt(jFormattedTextField1.getText());
+        }catch(NumberFormatException ex){
+            autoCompleteDelay = DEFAULT_AUTOCOMPLETE_DELAY_MS;

```

Comment:
I don't think we should silently set up default value if user provides incorrect data. We should loudly fail and tell user to provide correct data.